pkg/session/session.go

```diff
@@ -2274,6 +2276,20 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 	}
 
 	rs, err = s.Exec(ctx)
+
+	// check if is pipelined dml
+	if se.txn.Valid() && se.txn.IsPipelined() {
+		// Pipelined-DML can return assertion error and write conflict here, handle them like we handle
+		// errors returned from commit.
```

Comment:
Could you add comment to explain why need check here? 

The comment only says `can`, but doesn't explain `why`.